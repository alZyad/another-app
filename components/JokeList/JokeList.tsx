import { useState } from "react";
import data from "../../mockData.json";
import { JokeType } from "../../types/joke.types";
import Error from "../Error/Error";
import JokePreview from "../JokePreview/JokePreview";
import { StyleSheet, ScrollView, View, TextInput } from "react-native";

export default function JokeList() {
  const [searchTerm, setSearchTerm] = useState("");

  const { error, jokes }: { error: boolean; jokes: JokeType[] } = data;
  if (error) return <Error />;
  return (
    <View style={styles.container}>
      <TextInput style={styles.input} value={searchTerm} onChangeText={setSearchTerm} testID="searchInput" />
      <ScrollView contentContainerStyle={styles.listContent} style={styles.scrollList}>
        {jokes
          .filter((jokeData) =>
            ((jokeData?.joke || "") + (jokeData?.delivery || "") + (jokeData?.setup || ""))?.toLowerCase().includes(searchTerm.toLowerCase())
          )
          .map((jokeData) => (
            <JokePreview {...jokeData} key={jokeData.id} />
          ))}
      </ScrollView>
    </View>
  );
}

const styles = StyleSheet.create({
  input: {
    height: 40,
    margin: 12,
    borderWidth: 1,
    padding: 10,
  },
  listContent: {
    display: "flex",
    flexDirection: "row",
    flexWrap: "wrap",
    justifyContent: "center",
    gap: 25,
  },
  scrollList: {
    width: "100%",
  },
  container: {
    paddingTop: 30,
  },
});
