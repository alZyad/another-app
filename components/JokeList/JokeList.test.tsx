import React from "react";
import { render } from "@testing-library/react-native";
import JokeList from "./JokeList";

test("renders correctly", () => {
  const tree = render(<JokeList />).toJSON();
  expect(tree).toMatchSnapshot();
});
