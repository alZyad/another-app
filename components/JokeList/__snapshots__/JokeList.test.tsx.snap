// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`> JokeList renders correctly 1`] = `
<View
  style={
    {
      "paddingTop": 30,
    }
  }
>
  <TextInput
    onChangeText={[Function]}
    style={
      {
        "borderWidth": 1,
        "height": 40,
        "margin": 12,
        "padding": 10,
      }
    }
    testID="searchInput"
    value=""
  />
  <RCTScrollView
    contentContainerStyle={
      {
        "display": "flex",
        "flexDirection": "row",
        "flexWrap": "wrap",
        "gap": 25,
        "justifyContent": "center",
      }
    }
    style={
      {
        "width": "100%",
      }
    }
  >
    <View>
      <View
        style={
          {
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 50,
            "padding": 5,
            "width": 150,
          }
        }
        testID="jokePreview"
      >
        <Text>
          Hey wanna hear a
        </Text>
      </View>
      <View
        style={
          {
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 50,
            "padding": 5,
            "width": 150,
          }
        }
        testID="jokePreview"
      >
        <Text>
          Why is 6 afraid
        </Text>
      </View>
      <View
        style={
          {
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 50,
            "padding": 5,
            "width": 150,
          }
        }
        testID="jokePreview"
      >
        <Text>
          Ive got a really
        </Text>
      </View>
      <View
        style={
          {
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 50,
            "padding": 5,
            "width": 150,
          }
        }
        testID="jokePreview"
      >
        <Text>
          Programming is like sex
        </Text>
      </View>
      <View
        style={
          {
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 50,
            "padding": 5,
            "width": 150,
          }
        }
        testID="jokePreview"
      >
        <Text>
          A SQL statement walks
        </Text>
      </View>
      <View
        style={
          {
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 50,
            "padding": 5,
            "width": 150,
          }
        }
        testID="jokePreview"
      >
        <Text>
          Your mamas so FAT
        </Text>
      </View>
      <View
        style={
          {
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 50,
            "padding": 5,
            "width": 150,
          }
        }
        testID="jokePreview"
      >
        <Text>
          A guy walks into
        </Text>
      </View>
      <View
        style={
          {
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 50,
            "padding": 5,
            "width": 150,
          }
        }
        testID="jokePreview"
      >
        <Text>
          Java and C were
        </Text>
      </View>
      <View
        style={
          {
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 50,
            "padding": 5,
            "width": 150,
          }
        }
        testID="jokePreview"
      >
        <Text>
          Why did the web
        </Text>
      </View>
      <View
        style={
          {
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 50,
            "padding": 5,
            "width": 150,
          }
        }
        testID="jokePreview"
      >
        <Text>
          How many programmers does
        </Text>
      </View>
      <View
        style={
          {
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 50,
            "padding": 5,
            "width": 150,
          }
        }
        testID="jokePreview"
      >
        <Text>
          Why did the programmer
        </Text>
      </View>
      <View
        style={
          {
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 50,
            "padding": 5,
            "width": 150,
          }
        }
        testID="jokePreview"
      >
        <Text>
          Why are modern programming
        </Text>
      </View>
      <View
        style={
          {
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 50,
            "padding": 5,
            "width": 150,
          }
        }
        testID="jokePreview"
      >
        <Text>
          What do you get
        </Text>
      </View>
      <View
        style={
          {
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 50,
            "padding": 5,
            "width": 150,
          }
        }
        testID="jokePreview"
      >
        <Text>
          How do you know
        </Text>
      </View>
      <View
        style={
          {
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 50,
            "padding": 5,
            "width": 150,
          }
        }
        testID="jokePreview"
      >
        <Text>
          Two C strings walk
        </Text>
      </View>
      <View
        style={
          {
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 50,
            "padding": 5,
            "width": 150,
          }
        }
        testID="jokePreview"
      >
        <Text>
          Why do programmers confuse
        </Text>
      </View>
      <View
        style={
          {
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 50,
            "padding": 5,
            "width": 150,
          }
        }
        testID="jokePreview"
      >
        <Text>
          What do you call
        </Text>
      </View>
      <View
        style={
          {
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 50,
            "padding": 5,
            "width": 150,
          }
        }
        testID="jokePreview"
      >
        <Text>
          Java is like Alzheimers
        </Text>
      </View>
      <View
        style={
          {
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 50,
            "padding": 5,
            "width": 150,
          }
        }
        testID="jokePreview"
      >
        <Text>
          Why does no one
        </Text>
      </View>
      <View
        style={
          {
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 50,
            "padding": 5,
            "width": 150,
          }
        }
        testID="jokePreview"
      >
        <Text>
          Why did the Python
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
